test/model/store_test.dart

```diff
@@ -1082,6 +1082,22 @@ void main() {
         checkLastRequestFcm(token: '456def');
       }
     }));
+
+
+    test('fallback to default appBundleId in case packageInfo is null', ()async{
+      // This tests the case where packageInfo is null
+      // and default appBundleId gets passed to the [NotificationService.registerToken] method
+
+      prepareStore();
+      connection.prepare(json: {});
+
+      if(defaultTargetPlatform == TargetPlatform.iOS){
+        testBinding.packageInfoResult=null;
+        await NotificationService.registerToken(connection, token: '012abc');
+        checkLastRequestApns(token: '012abc', appid: 'com.zulip.flutter');
+        addTearDown(testBinding.reset);
```

Comment:
Usually this is called at the beginning of the test. `addTearDown` registers a callback function that gets called when the test is finished and "teared down", so there is no need to place it near the end.